tests/bsim/bluetooth/host/id/settings/test_scripts/settings.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/env bash
+# Copyright 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
+
+test_name='settings'
+test_exe="bs_${BOARD}_tests_bsim_bluetooth_host_id_${test_name}_prj_conf"
+simulation_id="id_${test_name}"
+verbosity_level=2
+EXECUTE_TIMEOUT=30
+
+cd ${BSIM_OUT_PATH}/bin
+
+if [ "${1}" != 'debug1' ]; then
+  Execute "./${test_exe}" \
+    -v=${verbosity_level} -s=${simulation_id} -d=0 -testid=dut1 \
+    -flash="${simulation_id}.log.bin"
```

Comment:
Both devices are using the same flash file. That sounds like trouble..
There is also no file to start with, so you may just want to keep the flash content in RAM?

Reply:
That's on purpose, I am testing that the settings stored are correct. To do that I simulate a device restart with a second device starting when the first one completed its work.
Tell me if I am wrong, but if I use the settings in RAM, I can't share the settings between devices.

Reply:
I see, the problem is that what you are doing it works when things align well enough. Meaning:
Both devices execute simultaneously, and both race to open the flash storage file.
During initialization, the NVMC model, if the file did not exist, or had zero size, will resize it to the correct size and clear it (to 0xFF).
In principle in between the opening of the file, and the check of its size, and the clearing of the content, the other device may have raced ahead of the other. Meaning, one device may open a file that is nonexistant / empty / right size but not cleared / partially cleared/ already contains settings. And the state is changing under its feet as things go. And one device may clear the content after the other one started writing settings to it.

If you know device 2 is the one reading the settings, and that device 1 will have written its settings in less than x ms, you can postpone the boot of device 2 by that amount to ensure the file is already created and its settings written.
But for that to work you would really need to hope that neither 1 or 2 modify the content anymore after.